analysis/src/analysis.js

```diff
@@ -66,6 +70,19 @@ class Analysis {
       }).catch(errorHandler);
     });
   }
+
+  _processNPMTask(attributes) {
+    return new Promise((resolve, reject) => {
+      Ana.log('analysis/processNextTask', taskAsString);
+
+      if (!attributes || !attributes.owner || !attributes.repo || !attributes.version || !attributes.npmPackage) {
+        errorHandler({retry: false, error: 'Task attributes missing or not a package'});
+        return;
+      }
+
+      var versionOrSha = attributes.sha ? attributes.sha : attributes.version;
```

Comment:
Fixed by actually implementing the rest of this. sha is ignored right now though, since it won't have been published on NPM as a sha.